:bug: Update the practitioner details to expect string resource Ids

dubdabasoduba opened the pull request:
fixes #https://github.com/opensrp/fhircore/issues/1995

dubdabasoduba committed:
:bug: Update the practitioner details to expect string resource Ids

dubdabasoduba committed:
:art: Code formatting via spotless apply

dubdabasoduba committed:
:arrow_up: Update versions

dubdabasoduba committed:
:art: Use static imports

dubdabasoduba commented:
Please review this too once you are done https://github.com/opensrp/hapi-fhir-jpaserver-starter/pull/56

ndegwamartin commented on practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java:
```diff
@@ -109,12 +112,11 @@ public PractitionerDetails getPractitionerDetails(
-                    && StringUtils.isNumeric(
-                            ((IBaseResource) practitioner).getIdElement().getIdPart())) {
+                    && StringUtils.isNumeric(practitioner.getIdElement().getIdPart())) {
                 practitionerId =
                         practitioner.getIdElement() != null
                                         && practitioner.getIdElement().getIdPart() != null
```
We already have the null check in the outer condition as well as the default of `0L`, we can refactor this line to just be   `practitionerId = practitioner.getIdElement().getIdPartAsLong();`

ndegwamartin commented on practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java:
```diff
@@ -126,15 +128,12 @@ public PractitionerDetails getPractitionerDetails(
                         practitioner.getIdElement() != null
                                         && practitioner.getIdElement().getIdPart() != null
-                                ? ((IBaseResource) practitioner)
-                                        .getIdElement()
-                                        .getIdPart()
-                                        .toString()
+                                ? practitioner.getIdElement().getIdPart()
                                 : "";
```
We already have the null check in the outer condition as well as the default of `""`, we can refactor this line to just be   `practitionerIdStringValue = practitioner.getIdElement().getIdPart();`

dubdabasoduba replied:
Makes sense, actually, I don't think we need to care about whether the id is a string or numeric.  https://github.com/opensrp/hapi-fhir/blob/148cbf119b3a90d129850ff19fc39ae3fdddc6c0/hapi-fhir-base/src/main/java/org/hl7/fhir/instance/model/api/IIdType.java#L50 returns a string which will work on search so we do not need to check whether the id is numeric or not

ndegwamartin commented on practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java:
```diff
@@ -109,12 +112,11 @@ public PractitionerDetails getPractitionerDetails(
             IBaseResource practitioner =
                     practitioners.size() > 0 ? practitioners.get(0) : new Practitioner();
             Long practitionerId = null;
-            String practitionerIdStringValue = null;
+            String practitionerIdStringValue;
```
We can set the defaults to `0L` and `""` respectively to coincide with the recommended refactors below

dubdabasoduba committed:
:bug: Update how we fetch the practitionerId.

Update how we fetch the practitionerId. We don't need to check whether the id is a String or Long.
